### PR TITLE
Return the same point in time in search response

### PR DESCRIPTION
**Pull request by dnhatn**
With this change, we will always return the same point in time in a search response as its input until we implement the retry mechanism for the point in times.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java [hunk 1]
```diff
@@ -543,8 +543,8 @@ protected final SearchResponse buildSearchResponse(InternalSearchResponse intern
             skippedOps.get(), buildTookInMillis(), failures, clusters, searchContextId);
     }
 
-    boolean includeSearchContextInResponse() {
-        return request.pointInTimeBuilder() != null;
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java:547]: Can you add a comment to explain that it is only used when creating the PIT ?

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
run elasticsearch-ci/packaging-sample-unix

File: server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java [hunk 2]
```diff
@@ -543,8 +543,10 @@ protected final SearchResponse buildSearchResponse(InternalSearchResponse intern
...
+    boolean buildPointInTimeFromSearchResults() {
+        // TODO: Until we implement the retry mechanism for point in times (i.e., replace an unavailable shard with an equivalent copy),
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java:547]: I noticed this TODO while working on an unrelated refactor. Do we need to address the TODO now that we retry point-in-times (#66713)?
